phoenix-core/src/main/java/org/apache/phoenix/iterate/SnapshotScanner.java

```diff
@@ -111,7 +111,11 @@ public SnapshotScanner(Configuration conf, FileSystem fs, Path rootDir,
   public Result next() throws IOException {
     values.clear();
     scanner.nextRaw(values);
-    statisticsCollector.collectStatistics(values);
+    // Needs to call collectStatistics even when values is empty.
+    // In this case, we pass hasMore with value 'false' so a new
+    // guide post will be created for the remaining cached data.
+    statisticsCollector.collectStatistics(values, ! values.isEmpty());
```

Comment:
Suggestion: another way of handling the last GPW is with the scanner close method as well or when trying to commit the GPW. With that approach, you won't need to explicitly add this boolean here.

Reply:
Adopted the suggestion